src/main/kotlin/io/kotlintest/KotlinTestEngine.kt

```diff
@@ -0,0 +1,220 @@
+package io.kotlintest
+
+import org.junit.platform.commons.JUnitException
+import org.junit.platform.commons.util.ReflectionUtils
+import org.junit.platform.engine.*
+import org.junit.platform.engine.discovery.PackageSelector
+import org.junit.platform.engine.support.hierarchical.EngineExecutionContext
+import java.lang.reflect.Modifier
+import java.util.*
+
+class KotlinTestContext(val executionRequest: ExecutionRequest) : EngineExecutionContext
+
+class KotlinTestEngine : TestEngine {
+
+  override fun execute(request: ExecutionRequest) {
+    println("Executing test request $request")
+    Project.beforeAll()
```

Comment:
No that we have an object which is a container with all specs, we should be able to harmonize the global interceptors with spec and test cases interceptors and use functions to wrap the whole project as with the interceptor chain in `SpecExecutor`. This would lift the whole interceptor mechanism on a new level of elegance and conceptual simplicity (through uniformity).